### PR TITLE
Dependencies field for all weight formats

### PR DESCRIPTION
**Pull request by FynnBe**

### CODE REVIEW

File: bioimageio/spec/model/v0_3/schema.py [hunk 1]
```diff
@@ -419,13 +419,13 @@ def raise_on_weights_format_mismatch(self, raw_node, **kwargs):
...
 
 
 class OnnxWeightsEntry(_WeightsEntryBase):
     bioimageio_description = "ONNX weights format"
     weights_format = fields.String(validate=field_validators.Equal("onnx"), required=True, load_only=True)
-    opset_version = fields.Integer()  # todo: required=True
+    opset_version = fields.Integer()
 
 
 class PytorchStateDictWeightsEntry(_WeightsEntryBase):
```
Comment (constantinpape) [inline bioimageio/spec/model/v0_3/schema.py:431]: Don't we need (optional) `pytorch_version` for the `PytorchStateDict` and `PytorchScript` entries as well?

File: bioimageio/spec/model/v0_4/schema.py [hunk 2]
```diff
@@ -166,9 +166,23 @@ def matching_halo_length(self, data, **kwargs):
             raise NotImplementedError(type(shape))
 
 
-class PytorchStateDictWeightsEntry(_WeightsEntryBase):
+class _WeightsEntryBase(_WeightsEntryBase03):
     raw_nodes = raw_nodes
+    dependencies = fields.Dependencies(
+        bioimageio_description="Dependency manager and dependency file, specified as `<dependency manager>:<relative "
```
Comment (constantinpape) [inline bioimageio/spec/model/v0_4/schema.py:172]: Can we extend the description here, to make clear that this is only for this weight format?

**Review by constantinpape — COMMENTED** [review]
Some minor comments.
But tests are still failing because something tries to cast string to int:
```
  bioimageio/spec/commands.py:111: in validate
      "warnings": ValidationWarning.get_warning_summary(validation_warnings),
  bioimageio/spec/shared/common.py:69: in get_warning_summary
      add_to_summary(summary, keys, msg)
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  s = {}, keys = [], msg = "missing 'pytorch_version'"
  
      def add_to_summary(s, keys, msg):
          key = keys.pop(0)
          if "[" in key:
              key, rest = key.split("[")
              assert rest[-1] == "]"
  >           idx = int(rest[:-1])
  E           ValueError: invalid literal for int() with base 10: 'torchscript'
  
  bioimageio/spec/shared/common.py:38: ValueError
```

File: bioimageio/spec/model/v0_4/raw_nodes.py [hunk 3]
```diff
@@ -45,26 +46,52 @@
...
+
+
 @dataclass
 class PytorchStateDictWeightsEntry(_WeightsEntryBase):
     weights_format_name = "Pytorch State Dict"
     architecture: ImportableSource = missing
     architecture_sha256: Union[_Missing, str] = missing
-    dependencies: Union[_Missing, Dependencies] = missing
     kwargs: Union[_Missing, Dict[str, Any]] = missing
+    pytorch_version: Union[_Missing, distutils.version.StrictVersion] = missing
```
Comment (FynnBe) [inline bioimageio/spec/model/v0_4/raw_nodes.py:78]: pytorch_version for PytorchStateDict

File: bioimageio/spec/model/v0_4/raw_nodes.py [hunk 4]
```diff
@@ -45,26 +46,52 @@
...
+
+@dataclass
+class TensorflowSavedModelBundleWeightsEntry(_WeightsEntryBase, TensorflowSavedModelBundleWeightsEntry03):
+    pass
 
 
 @dataclass
 class TorchscriptWeightsEntry(_WeightsEntryBase):
     weights_format_name = "Torchscript"
+    pytorch_version: Union[_Missing, distutils.version.StrictVersion] = missing
```
Comment (FynnBe) [inline bioimageio/spec/model/v0_4/raw_nodes.py:94]: pytorch_version for torchscript node...

File: bioimageio/spec/model/v0_3/schema.py (hunk 1 above)
Comment (FynnBe) [reply bioimageio/spec/model/v0_3/schema.py:431]: yes, see my comments below..

File: bioimageio/spec/model/v0_4/schema.py (hunk 2 above)
Comment (FynnBe) [reply bioimageio/spec/model/v0_4/schema.py:172]: I added: "These dependencies are only valid in the context of the specified weight format."

File: bioimageio/spec/model/v0_4/schema.py [hunk 5]
```diff
@@ -166,9 +166,24 @@ def matching_halo_length(self, data, **kwargs):
...
+        "path to file>`. For example: 'conda:./environment.yaml', 'maven:./pom.xml', or 'pip:./requirements.txt'. "
+        "These dependencies are only valid in the context of the specified weight format."
```
Comment (constantinpape) [inline bioimageio/spec/model/v0_4/schema.py:174]:
```suggestion
        "These dependencies are only used for the specified weight format."
```

Comment (constantinpape) [reply bioimageio/spec/model/v0_4/schema.py:174]: I think "used for" is shorter and more accurate here. But feel free to stick with your original formulation if you prefer it.

File: bioimageio/spec/model/v0_3/schema.py (hunk 1 above)
Comment (constantinpape) [reply bioimageio/spec/model/v0_3/schema.py:431]: Ok, I see. These are the files for 0.3...

**Review by constantinpape — APPROVED** [review]
I left one suggestion to reformulate the doc string. Otherwise this looks good to me, feel free to merge when you think this is ready @FynnBe

**Pull request merged**